compose-state-events/src/main/java/de/palm/composestateevents/NavigationEventEffect.kt

```diff
@@ -0,0 +1,40 @@
+package de.palm.composestateevents
+
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+
+/**
+ * Other than the regular [EventEffect] this one will
+ * consume the event before actually executing the action.
+ */
+@Composable
+fun <T> NavigationEventEffect(
+    event: StateEventWithContent<T>,
+    onConsumed: () -> Unit,
+    action: suspend (T) -> Unit
+) {
+    LaunchedEffect(key1 = event, key2 = onConsumed) {
+        if (event is StateEventWithContentTriggered<T>) {
+            onConsumed()
+            action(event.content)
+        }
+    }
+}
+
+/**
+ * Other than the regular [EventEffect] this one will
```

Comment:
The comments for these two composables are exactly the same. Please make shure to document which of these 2 methods to use in which case (with an argument and without).